src/back/spv/writer.rs

```diff
@@ -1384,17 +1371,36 @@ impl Writer {
             }
             crate::Expression::GlobalVariable(handle) => self.global_variables[handle.index()].id,
             crate::Expression::Constant(handle) => self.constant_ids[handle.index()],
+            crate::Expression::Splat { size, value } => {
+                let value_id = self.cached[value];
+                self.temp_chain.clear();
+                self.temp_chain.resize(size as usize, value_id);
+
+                let id = self.id_gen.next();
+                block.body.push(Instruction::composite_construct(
+                    result_type_id,
+                    id,
+                    &self.temp_chain,
+                ));
+                id
+            }
             crate::Expression::Compose {
                 ty: _,
                 ref components,
             } => {
-                //TODO: avoid allocation
-                let mut constituent_ids = Vec::with_capacity(components.len());
+                self.temp_chain.clear();
                 for &component in components {
```

Comment:
holy magic! Never used splice before

Reply:
I tried it, and I haven't reached a solid conclusion on whether it's good to use here. First, because it's fairly magical: there is a range argument, a closure, and it returns an iterator. Its guarantees on performance depend on how it's used... Second, because `self.cached` can't be used inside the closure, so that's another line to spend, anyway.
Sometimes the dumb code is the best :)

Reply:
Oh, that's too bad: the borrow checker doesn't see how to split the borrow, so the closure borrows all of `self`. This does work:
```
self.temp_list.clear();
let cached = &self.cached;
self.temp_list.extend(components.iter().map(|&c| cached[c]));
```
but at that point it's becoming arguable whether this is a clarification or not. Oh well.

Reply:
yes, it's certainly workaroundable, just a bit less nice than originally desired :)